src/plugins/discover/public/application/components/discover_grid/discover_grid_flyout.tsx

```diff
@@ -29,27 +31,62 @@ import { getContextUrl } from '../../helpers/get_context_url';
 interface Props {
   columns: string[];
   hit: ElasticSearchHit;
+  hits?: ElasticSearchHit[];
   indexPattern: IndexPattern;
   onAddColumn: (column: string) => void;
   onClose: () => void;
   onFilter: DocViewFilterFn;
   onRemoveColumn: (column: string) => void;
   services: DiscoverServices;
+  setExpandedDoc: (doc: ElasticSearchHit) => void;
 }
 
+type ElasticSearchHitWithRouting = ElasticSearchHit & { _routing?: string };
+
+function getDocFingerprintId(doc: ElasticSearchHitWithRouting) {
+  const routing = doc._routing || '';
+  return [doc._index, doc._id, routing].join('||');
+}
+
+function getIndexByDocId(hits: ElasticSearchHit[], id: string) {
+  return hits.findIndex((h) => {
+    return getDocFingerprintId(h) === id;
+  });
+}
 /**
  * Flyout displaying an expanded Elasticsearch document
  */
 export function DiscoverGridFlyout({
   hit,
+  hits,
   indexPattern,
   columns,
   onFilter,
   onClose,
   onRemoveColumn,
   onAddColumn,
   services,
+  setExpandedDoc,
 }: Props) {
+  const pageCount = useMemo<number>(() => (hits ? hits.length : 0), [hits]);
+  const activePage = useMemo<number>(() => {
+    const id = getDocFingerprintId(hit);
+    if (!hits || pageCount <= 1) {
```

Comment:
won't hits always be an empty array?

Reply:
currently there's `rows` assigned in `DiscoverGrid`, which can be undefined, but I intend to change this when finishing the document selector 👍 should be an empty array soon